diagrams/cs-for-hackers.js

```diff
@@ -0,0 +1,117 @@
+var nameSpace = 'template';
+
+var svgWidth  = 250;
+var svgHeight = 250;
+
+var xOffset = 40;
+
+
+// dynamic values
+
+var textColor = '#555555';
+
+//table appearance
+var rowHeight = 30;
+var rowWidth = 110;
+var numRows = 6;
+// var rowColor = '#eff9fc';
+var strokeColor = '#555555';
+
+var tableContent = [];
+var lightOutlineRows = []; // range of rows to outline lightly
+var darkOutlineRows = []; // range of rows to outline darkly
+
+// functions for drawing diagrams
+
+function drawTable(tableNumber, lightOutlineRows = [], darkOutlineRows = []) {
+  fillTable(tableNumber);
+
+  for (i = 1; i < numRows+1; i++) { 
+      // draw row numbers
+      generateText(i - 1, 20, i*rowHeight + 19, 13, null, textColor);
+      // draw rows
+      if (lightOutlineRows.length == 1 && lightOutlineRows[0] == i) {
+        generateRect(xOffset, i*rowHeight, rowWidth, rowHeight, 0, null, strokeColor, 2);
+      } else if (darkOutlineRows.length == 1 && darkOutlineRows[0] == i) {
+        generateRect(xOffset, i*rowHeight, rowWidth, rowHeight, 0, null, strokeColor, 3);
+      } else {
+        generateRect(xOffset, i*rowHeight, rowWidth, rowHeight, 0, null, strokeColor, 1);
+      }
+      generateText(tableContent[i], rowWidth - 15, i*rowHeight + 19, 13, null, textColor);
+      // if (i == numRows) {
+        // generateMask(100, 500, rowHeight, true);
+      // }
+  }
+}
+
+function fillTable(tableNumber) {
+  switch(tableNumber) {
+    case 1:
+      for (i = 1; i < numRows+1; i++) { 
+        tableContent[i] = '0 0 0 0 0 0 0 0';
+      }
+      break;
+
+    case 2:
+      tableContent[1] = '0 0 0 0 0 0 0 0';
+      tableContent[2] = '0 0 0 0 0 0 0 1';
+      tableContent[3] = '0 0 0 0 0 0 1 1';
+      tableContent[4] = '0 0 0 0 0 0 1 1';
+      tableContent[5] = '0 0 0 0 0 1 0 0';
+      break;
+
+    case 3:
+      tableContent[1] = '0 0 0 0 0 0 0 0';
+      tableContent[2] = '0 0 0 0 0 0 0 0';
+      tableContent[3] = '0 0 0 0 0 1 1 0';
+      tableContent[4] = '0 0 0 0 0 0 0 0';
+      tableContent[5] = '0 0 0 0 0 0 0 0';
+      break;
+  }
+}
+
+
+// generate diagrams
+
+var diagramFunctions = [
+
+    function drawTable1() {
```

Comment:
make these a little more descriptive, we should kind of know what the diagram is by its name. see SVG_DATA in settings.py for examples